qiita_core/testing.py

```diff
@@ -0,0 +1,35 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from json import loads
+from time import sleep
+
+from moi import r_client
+
+from qiita_db.processing_job import ProcessingJob
+
+
+def wait_for_prep_information_job(prep_id, raise_if_none=True):
+    res = r_client.get('prep_template_%d' % prep_id)
+
+    if raise_if_none:
+        assert res is not None, "unexpectedly None"
```

Comment:
This may be overcautious, but what about raising an actual AssertionError, when we use `assert` we run the risk that if python is being executed in an "optimized" mode, these statements will be ignored. I suggest this be changed to:

```python
if raise_if_none and res is None
    raise AssertionError('Unexpectedly None')
```